### PR TITLE
  Move the image file detection code to the File class

### PR DESCRIPTION
**Pull request by asliturk**

### CODE REVIEW

**Comment by Dexesttp** [comment]
It's way better than the way it was done before ! (side note, you probably want to pass a `const StringView&` instead of a `String&` as argument. That would simplify things ^^)

However, the list of allowed extensions / formats is still is duplicated between your new `LibCore/File.cpp#is_image_file(String& path)` method and the [`LibGfx/Bitmap.cpp#load_from_file(const StringView& path)`](https://github.com/SerenityOS/serenity/blob/master/Libraries/LibGfx/Bitmap.cpp#L87) method. And they have to be kept in sync for most of the logic to work properly.

While it seems kind of weird to put another file-extension-detecting method in Bitmap, maybe moving the method to `LibGfx::Bitmap::is_path_a_supported_image_format(const StringView&)` (or something similar) would mitigate that ?
This would also allow us to use a `#define ENUMERATE_IMAGE_FORMATS __ENUMERATE_IMAGE_FORMAT(load_gif, .gif)`-like macro for matching the file formats when parsing and reading (which would make keeping both of them in sync very easy).

**Comment by alimpfard** [comment]
`LibMagic` time?

**Comment by bugaevc** [comment]
> LibCore: File, Add new is_image_file method

Please see https://github.com/SerenityOS/serenity/pull/2550#issuecomment-643572519 :slightly_smiling_face: — the commit message should be "LibCore: Add a new `File::is_image_file()` method"

File: Libraries/LibCore/File.h [hunk 1]
```diff
@@ -38,6 +38,15 @@ class File final : public IODevice {
     virtual ~File() override;
 
     static Result<NonnullRefPtr<File>, String> open(const String& filename, IODevice::OpenMode, mode_t = 0644);
+    static bool is_image_file(const String& path)
```
Comment (bugaevc) [inline Libraries/LibCore/File.h:41]:
```suggestion
    static bool is_image_file(const StringView& path)
```

**Comment by asliturk** [comment]
Hi @Dexesttp,

It is very good recommendation. I moved the new method to the `Gfx:.Bitmap` class as you suggested.

**Pull request merged**